source/developer/bot-accounts.rst

```diff
@@ -0,0 +1,186 @@
+
+Command Line Interface (CLI)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can use the following CLI command to convert an existing user account to a bot:
+
+.. code-block:: text
+
+  user convert user@example.com --bot
+
+In addition to email, you may identify the user by its username or user id.
+
+Bot accounts which were converted from user accounts will have their authentication data cleared if they were email/password accounts. Those synced from LDAP/SAML will not have their authentication data cleared so that LDAP/SAML sync performs correctly.
+
+Plugins
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Plugins can create bot accounts through an ``EnsureBot`` helper function. For an example, see `the Demo Plugin <https://github.com/mattermost/mattermost-plugin-demo/blob/master/server/configuration.go#L189>`_.
+
+Bots created by a plugin use the plugin's ID as the creator, unless otherwise specified by the plugin.
+
+Technical Notes
+-----------------------
+
+Data Model
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Each bot account has a row in the **Users** table. The entry is tied to a row in the **Bots** table by a ``UserId`` column.
+
+The Bots table schema is described as follows:
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "UserId", "User ID of the bot user", "string", "Y"
+    "Username", "Username of the bot account", "string", "Y"
+    "DisplayName", "Display name of the bot account", "string", "N"
+    "Description", "Description of the bot account", "string", "N"    
+    "OwnerId", "User ID of the owner of the bot", "string", "Y"
+    "CreateAt", "Unix timestamp of creation time", "int64", "Y"
+    "UpdateAt", "Unix timestamp of update time", "int64", "Y"
+    "DeleteAt", "Unix timestamp of deletion time", "int64", "Y"
+
+The row in the Users table is used to track the bot’s other data, such as username.
+
+Frequently Asked Questions
+-----------------------------
+
+Should I migrate all my integrations to use bot accounts?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For your integrations using RESTful API and plugins, yes. To do so, you can either convert an existing account to a bot, or create a new bot account using the steps outlined above.
+
+Once you create a bot account, use the generated token to access the RESTful API on behalf of a bot and interact in the Mattermost server.
+
+For your webhook and slash command integrations, you cannot migrate them to use bot accounts, as they require a user account at this time. However, an option is to migrate the webhooks or slash commands to a plugin, which in turn can use bot accounts.
+
+How do I convert an existing account to a bot account?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Use the following CLI command to convert an existing user account to a bot:
+
+.. code-block:: text
+
+  user convert user@example.com --bot
```

Comment:
```suggestion
  user convert --bot user@example.com
```

Reply:
@crspeller Are you sure? In the source code, we give the example as `user convert user@example.com anotherUser --bot`

https://github.com/mattermost/mattermost-server/blob/master/cmd/mattermost/commands/user.go#L48

Reply:
Okay, I'll leave it as is then to be consistent with other documented CLI commands and the example given in commands/user.go

If it's conventional to do flags first, then maybe that's something we should update throughout